### PR TITLE
Add SendMessageTimeout

### PR DESCRIPTION
**Pull request by adrianrus**

### CODE REVIEW

File: src/User32/User32.cs [hunk 1]
```diff
@@ -647,6 +647,37 @@ public static partial class User32
...
+        /// <param name="pdwResult">The result of the message processing. The value of this parameter depends on the message that is specified.</param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        /// SendMessageTimeout does not provide information about individual windows timing out if HWND_BROADCAST is used.
+        /// If the function fails or times out, the return value is 0.
+        /// To get extended error information, call GetLastError.
+        /// If GetLastError returns ERROR_TIMEOUT, then the function timed out.
+        /// </returns>
+        [DllImport(nameof(User32), SetLastError = true)]
+        public static extern IntPtr SendMessageTimeout(HandleRef hWnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, SendMessageTimeoutFlags flags, int timeout, out IntPtr pdwResult);
```
Comment (AArnott) [inline src/User32/User32.cs]: Please use `IntPtr` for HWNDs instead of the obsolete `HandleRef` class.

**Review by AArnott — CHANGES_REQUESTED** [review]
This looks *really* good. You've obviously taken a lot of care to follow the guidelines for this project. Thank you.

Just one code change, and then to fix the build break you should use the automated code fix offered in Visual Studio when you double-click the error in the error list then click the light bulb. You should be able to fix all the build errors at once by clicking to fix all in project.

File: src/User32/User32+SendMessageTimeoutFlags.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+namespace PInvoke
+{
+    /// <content>
+    /// Contains the <see cref="SendMessageTimeoutFlags"/> nested type.
+    /// </content>
+    public partial class User32
+    {
+        /// <summary>
+        /// Possible flag values for <see cref="User32.SendMessageTimeout"/>
+        /// Documentation pulled from MSDN.
```
Comment (vbfox) [inline src/User32/User32+SendMessageTimeoutFlags.cs]: Nitpick: That can be removed, we pull all documentation from MSDN :)

**Review by vbfox — APPROVED** [review]

**Comment by AArnott** [comment]
The latest build break is due to the appveyor build agents updating to 15.3. I'll prepare a fix for that then rebuild your PR.

File: src/User32/User32+SendMessageTimeoutFlags.cs (hunk 2 above)
Comment (AArnott) [reply src/User32/User32+SendMessageTimeoutFlags.cs]: I removed it.

**Review by AArnott — APPROVED** [review]

**Comment by AArnott** [comment]
Thanks, @adrianrus

**Comment by adrianrus** [comment]
Thanks for fixing the build! @AArnott

**Pull request merged**